OpenOversight/app/main/views.py

```diff
@@ -416,6 +416,19 @@ def edit_department(department_id):
                 if rank:
                     new_ranks.append((rank, order))
                     order += 1
+            updated_ranks = form.jobs.data
+            if len(updated_ranks) < len(original_ranks):
+                deleted_ranks = [rank for rank in original_ranks if rank.job_title not in updated_ranks]
+                for rank in deleted_ranks:
+                    if Assignment.query.filter(Assignment.job_id.in_([rank.id for rank in deleted_ranks])).count() == 0:
+                        db.session.delete(rank)
+                        db.session.commit()
```

Comment:
Should the `db.session.commit()` be lifted out of the for loop and simply placed before the function returns or exits? That way it's an all-or-nothing deal and the user can't get stuck with partial or inconsistent results. I don't know what best practices are here, but this makes sense to me.

Reply:
Good point, I refactored and the `db.session.commit()` is now outside of the loop.